packages/core/src/app/shipping/Shipping.tsx

```diff
@@ -109,11 +111,18 @@ class Shipping extends Component<ShippingProps & WithCheckoutShippingProps, Ship
             loadShippingOptions,
             onReady = noop,
             onUnhandledError = noop,
+            cartHasPromotionalItems,
+            isMultiShippingMode,
+            isNewMultiShippingUIEnabled,
         } = this.props;
 
         try {
             await Promise.all([loadShippingAddressFields(), loadShippingOptions(), loadBillingAddressFields()]);
 
+            if(cartHasPromotionalItems && isMultiShippingMode && isNewMultiShippingUIEnabled) {
+                await this.handleMultiShippingModeSwitch();
+            }
```

Comment:
Moving forward, we should avoid putting heavy API calls function inside a `useEffect` hook or similar places like this for better testability.
Instead, it would be more effective to put this function call on the button that triggers it.
This will simplify our testing, as we can focus on testing the button itself rather than preparing the entire component for testing an API-related functionality.
cc @animesh1987

Reply:
The button that triggers this is going to be on Cart page, hence that cannot be controlled.